extension-browser.webpack.config.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+const path = require('path')
+const CopyWebpackPlugin = require('copy-webpack-plugin')
+const { DefinePlugin } = require('webpack')
+
+module.exports = {
+  entry: {
+    extension: './src/extension.ts'
+  },
+  mode: 'none', // this leaves the source code as close as possible to the original (when packaging we set this to 'production')
+  target: 'webworker', // extensions run in a webworker context
+  resolve: {
+    mainFields: ['browser', 'module', 'main'],
+    extensions: ['.ts', '.js'], // support ts-files and js-files
+    fallback: {
+      'fs': false, // do not include a polyfill,
+      'assert': false,
+      'unxhr': false,
+      'glob': false,
+      'http': false,
+      'https': false,
+      'url': false,
+      'zlib': false,
+      'os': false,
+      'child_process': false,
+      'crypto': false,
+      'stream': false,
+      'path': require.resolve('path-browserify'),
+      'util': require.resolve('util')
+    }
+  },
+  module: {
+    rules: [{
+      test: /\.ts$/,
+      exclude: /node_modules/,
+      use: [{
+        // configure TypeScript loader:
+        // * enable sources maps for end-to-end source maps
+        loader: 'ts-loader',
+        options: {
+          compilerOptions: {
+            'sourceMap': true,
+          },
+          configFile: 'tsconfig.browser.json'
+        }
+      }]
+    }]
+  },
+  externals: {
+    'vscode': 'commonjs vscode', // ignored because it doesn't exist,
+    'applicationinsights-native-metrics': 'commonjs applicationinsights-native-metrics', // ignored because we don't ship native module
+    '@opentelemetry/tracing': 'commonjs @opentelemetry/tracing' // ignored because we don't ship this module
+  },
+  performance: {
+    hints: false
+  },
+  output: {
+    // all output goes into `dist`.
+    // packaging depends on that and this must always be like it
+    filename: '[name].js',
+    path: path.join(__dirname, 'dist', 'browser'),
+    libraryTarget: 'commonjs',
+  },
+  amd: false, // disable amd
+  // yes, really source maps
+  devtool: 'source-map',
+  plugins: [
```

Comment:
We might want to add `IgnorePlugin` to reduce the size of the bundle: https://webpack.js.org/plugins/ignore-plugin/

Reply:
Alas, I do not understand how we could use this.

`extension.js` seems to be about 4 MiB and  `extension.js.map` which presumably we should only produce for development is about 4.8 MiB on my machine.

Reply:
We are using a lot of polyfills and some features won't be available in a web environment but the code will still be included in the bundle.
My idea is to use the `IgnorePlugin` to make sure that we only include what we need.
But we can do that at a later point.